agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -49,17 +66,16 @@ def on_dns_add(topic, event)
     # @param [Docker::Event] event
     def on_container_event(topic, event)
       if event.status == 'start'
-        container = Docker::Container.get(event.id) rescue nil
-        if container
-          self.start_container(container)
-        else
-          warn "skip start event for missing container=#{event.id}"
-        end
-      elsif event.status == 'restart'
         if network_adapter.router_image?(event.from)
+          info "Restart after weave restart"
+
           wait_weave_running?
 
-          self.start
+          self.start_all_containers
```

Comment:
Note how the `on_container_event` now calls `start_all_containers`, instead of `start`, to avoid the #2225 re-`subscribe` issue.